### PR TITLE
Use proper CSP headers for Overlay

### LINKED ISSUE #18187: Fix possible problem with CSP in overlay feature
See https://demo.matomo.cloud/index.php?module=Overlay&period=month&date=today&idSite=1#?l=https$3A$2F$2Fdivezone.net$2Fdiving$2Findonesia$3F__prod_demo$3D1$26__enterprise_demo_idsite$3D1

seeing this error 
image

When we remove the report only mode in https://github.com/matomo-org/matomo/issues/17923 this could break the overlay feature.

Not really sure what the problem is there?

### PR DESCRIPTION
**Pull request by sgiehl**
### Description:

fixes #18187

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: plugins/Overlay/Controller.php [hunk 1]
```diff
@@ -63,6 +65,7 @@ public function index()
         $view->ssl = ProxyHttp::isHttps();
         $view->siteUrls = SitesManager\API::getInstance()->getSiteUrlsFromId($this->site->getId());
 
+        StaticContainer::get(SecurityPolicy::class)->allowEmbedPage();
```
Comment (tsteur) [inline plugins/Overlay/Controller.php]: @sgiehl in controller can directly use `$this->securityPolicy->...`

**Review by justinvelluppillai — APPROVED** [review]
Removed unneeded imports and now looks good to merge

**Comment by sgiehl** [comment]
@tsteur Might be good if we can deploy this one to cloud before 4.6.0 is released to be sure the overlay would work correct again. Otherwise it could break completely if CSP will be out of report only in 4.6.0

**Comment by tsteur** [comment]
Was deployed last week so it's already on the demo @sgiehl

**Pull request merged**